app/post.go

```diff
@@ -1296,10 +1341,102 @@ func (a *App) convertUserNameToUserIds(usernames []string) []string {
 	return usernames
 }
 
+// GetLastAccessiblePostTime returns CreateAt time of last accessible post as per the cloud limit
+func (a *App) GetLastAccessiblePostTime(useCache bool) (int64, *model.AppError) {
+	cfg := a.Config()
+	if cfg.FeatureFlags == nil || !cfg.FeatureFlags.CloudFree {
+		return 0, nil
+	}
+
+	if useCache {
```

Comment:
Interestingly, I initially implemented the way you're suggesting🙂.
But then it felt a little off to me, as the app was now depending on a "helper" piece, in this "cache" to work correctly.

I generally try to make sure that app should keep running as long as its "core" features/logic are working. So, in this case even if the "cached" value doesn't work, the app is still able to work(even though slower), while the cache can get fixed.

And for the same reason, I logged the errors instead of returning as the core functionality is still working, so the app should also keep running.
Maybe I can change the Warn to Err, and I'm not sure, how these logs work on Prod, especially the .Err() logs. Does admin get notifications or has an access to some UI to see these error logs?

Also, I think it's a common pattern to fetch from the cache > if not found, fetch from the main source > then update the cache > and return the data.

Let me know your thoughts, I can revert it, if you feel we should go that way.

Reply:
> I generally try to make sure that app should keep running as long as its "core" features/logic are working. So, in this case even if the "cached" value doesn't work, the app is still able to work(even though slower), while the cache can get fixed.

Yep, do all of that, but the method names should be separate, and there is no need for the `useCache` flag.

> And for the same reason, I logged the errors instead of returning as the core functionality is still working, so the app should also keep running.

I understand your intention, but in practice this leads to more harm than help. You want to keep the app running, but the app is already failing, and all we are doing is hiding the errors from the user. This doesn't really help and the user gets confused because no error is surfaced.

So here is my proposal:

App method: `GetLastAccessiblePostTime()`. (no extra param needed)
- check cache
- if found return
- if not found, compute and insert in cache. 
 
Job method: `ComputeLastAccessiblePostTime()`:
- compute and insert in cache.

What I was suggesting is that to refactor these small bits into separate small methods so that they can be called from these 2 high level methods. E.g. the "compute and insert in cache" could be its own method.

WDYT?

Reply:
SGTM.